contracts/pegged/PeggedTokenBridge.sol

```diff
@@ -50,11 +58,16 @@ contract PeggedTokenBridge {
         address _token,
         uint256 _amount,
         uint64 _nonce
-    ) external {
-        bytes32 id = keccak256(abi.encodePacked("burn", _token, msg.sender, _amount, _nonce));
-        require(records[id] == false, "record exists");
-        records[id] = true;
+    ) external whenNotPaused {
+        bytes32 burnId = keccak256(abi.encodePacked(msg.sender, _token, _amount, _nonce, uint64(block.chainid)));
```

Comment:
Note that this can match a `mintId`, but maybe that's fine given the nonce is user-specified.

Reply:
mintId's hashed data length is 20+20+32+8+32 = 112, while burnId's hashed data length is 20+20+32+8+8 = 88.